### ISSUE TITLE
Atomic Components Implementation v2

### ISSUE THREAD

**Issue by ajmnz** [benawad/dogehouse#698]
<h1>Atomic Components v2</h1>

This is the follow-up for #637.

### Ready to be implemented

- [x] [Taken] RoomCard - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1355%3A831
    - Requires ScheduledTime, IconHeading & Tag ✅
- [x] [Taken] UserSummaryCard - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1356%3A771
    - Requires UserAvatar, UserBadge ✅
- [x] [Taken] GlobalSearch - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1356%3A949
    - Requires SearchBar, SearchHistory, SearchResult, SearchOverlay ✅
- [x] [Taken] SettingsDropdown - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1397%3A46
    - Requires SettingsIcon, BaseOverlay ✅
- [x] [Taken] NotificationsDropdown - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1397%3A143
    - Requires BaseOverlay, NotificationElement ✅
- [x] [Taken] MessagesDropdown - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1397%3A201
    - Requires BaseOverlay, MessageElement ✅
- [x] [Taken] MinimizedRoomCard - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1356%3A773
    - Requires Button, BoxedIcon ✅
- [x] [Taken] UpcomingRoomsCard - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1356%3A878
    - Requires BoxedIcon ✅

### Top level components

- [x] [Taken] FriendsOnline - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1397%3A474
    - Requires UserAvatar ✅
- [x] [Taken] Feed - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1397%3A797
    - Requires Button, RoomCard ✅
- [x] [Taken] ProfileBlock - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1397%3A329
    - Requires UserSummaryCard, MinimizedRoomCard, UpcomingRoomsCard ❌
- [x] [Taken] Header - https://www.figma.com/file/CS01VVLR7ArQl0afYFkNj3/Web-App?node-id=1397%3A894
    - Requires GlobalSearch, Button, UserAvatar ✅

----

**Don't add any margin around the components.**

Put your component in `kibbeh/src/ui` and create a story in `kibbeh/src/stories`

Use tailwind for styling (see the tailwind config for theme info).

You have to use TypeScript.

Work on the `new-design` branch.

Please comment below if you start to work on a component, but only claim a component if you plan to work on it right away so it doesn't block other people.

Assets: https://github.com/benawad/dogehouse/tree/new-design/.redesign-assets

If you have any questions ask Ben or @ ajb on [Discord](https://discord.gg/82HzQCJCDg).

**Comment by sahilmob** [comment]
I will work on UserSummaryCard

**Comment by lenilsondc** [comment]
I'm working on `RoomCard`

**Comment by overlisted** [comment]
I will do `MinimizedRoomCard` tomorrow

**Comment by sahilmob** [comment]
I will do GlobalSearch

**Comment by tejasag** [comment]
I will do UpcomingRoomsCard

**Comment by andrewroblesdev** [comment]
I got `SettingsDropdown`

**Comment by tejasag** [comment]
`NotificationsDropdown` is on me!

**Comment by jee1mr** [comment]
I will do `MessagesDropdown`

**Comment by jee1mr** [comment]
@ajmnz @andrewrobles @tejasag What are these? Are they already done?
<img width="776" alt="Screenshot 2021-03-12 at 9 08 36 PM" src="https://user-images.githubusercontent.com/6749387/110962662-35175300-8377-11eb-90d7-f558d1ed8d5f.png">

<img width="776" alt="Screenshot 2021-03-12 at 9 08 40 PM" src="https://user-images.githubusercontent.com/6749387/110962645-334d8f80-8377-11eb-848d-50eab03c9056.png">

<img width="776" alt="Screenshot 2021-03-12 at 9 08 43 PM" src="https://user-images.githubusercontent.com/6749387/110962631-30529f00-8377-11eb-9f0d-f81c75796db1.png">

**Comment by PrinceRajRoy** [comment]
I will do `UpcomingRoomsCard`

**Comment by andrewroblesdev** [comment]
@jee1mr Looks like whoever created the `BaseOverlay` component was kind enough to provide a `Settings` example. My understanding is that my task is to create the `SettingsDropdown.tsx` component and `SettingsDropdown.story.tsx` story which is  currently in progress. Does that make sense?

**Comment by jee1mr** [comment]
@andrewrobles Yes, got it!

**Comment by jee1mr** [comment]
I will do FriendsOnline as well.

**Comment by Mvrs** [comment]
Which one is open? 😅

**Comment by jee1mr** [comment]
> Which one is open? 😅

All taken. Wait for Feed, ProfileBlock, and Header to open up.

**Comment by yvesnrb** [comment]
> > Which one is open? sweat_smile
> 
> All taken. Wait for Feed, ProfileBlock, and Header to open up.

Can I preemptively claim Feed? It seems like RoomCard will be merged soon.

**Comment by itsjeremyrivera** [comment]
Can I get started on the header Component?

**Comment by itsjeremyrivera** [comment]
Or any next one that will be open?

**Comment by ajmnz** [comment]
> @ajmnz @andrewrobles @tejasag What are these? Are they already done?

Yep, the one that did `BaseOverlay` also added what it'd look like with every variant. But that's just the story, the component needs to be implemented.

**Comment by ajmnz** [comment]
> @ajmnz @andrewrobles @tejasag What are these? Are they already done?

Yep, the one that did `BaseOverlay` also added what it'd look like with every variant. But that's just the story, the component needs to be implemented.

**Comment by yvesnrb** [comment]
> > > Which one is open? sweat_smile
> > 
> > 
> > All taken. Wait for Feed, ProfileBlock, and Header to open up.
> 
> Can I preemptively claim Feed? It seems like RoomCard will be merged soon.

Working on Feed now. RoomCard just got merged.

**Comment by Yawanseh** [comment]
@ajmnz is there anything open here?
I guess it should be moved to the `someone is working on it` column

**Comment by ajmnz** [comment]
> @ajmnz is there anything open here?
> I guess it should be moved to the `someone is working on it` column

`ProfileBlock`is yet to be claimed.

**Comment by ice-rahul** [comment]
I will do `ProfileBlock`

**Comment by ajmnz** [comment]
> Can I get started on the header Component?

I assumed you claimed it, but you'll have to wait until `GlobalSearch` is merged.

**Comment by ajmnz** [comment]
@JeremyRivera `GlobalSearch` has been merged, you can start on `Header` if you wish

**Comment by overlisted** [comment]
Is the next step going to be the big layout itself? Who's going to do it?

**Comment by ajmnz** [comment]
> Is the next step going to be the big layout itself? Who's going to do it?

Maybe we'll wait until tablet & phone mockups are done, I'm not sure

**Comment by kprocks86** [comment]
@ajmnz can you reconfirm with everyone who has claimed the components, that they are still working on it. or else few more people in line can start.

**Comment by ajmnz** [comment]
> @ajmnz can you reconfirm with everyone who has claimed the components, that they are still working on it. or else few more people in line can start.

Everything is taken

**Comment by benawad** [comment]
@PrinceRajRoy are you still working on `UpcomingRoomsCard`?

**Comment by benawad** [comment]
@JeremyRivera are you working on `Header`?

**Comment by PrinceRajRoy** [comment]
@benawad, opening a PR

**Comment by yvesnrb** [comment]
I can take on header if it has been abandoned.

**Comment by ice-rahul** [comment]
waiting for `UpcomingRoomsCard` to get `ProfileBlock` unblocked

**Comment by benawad** [comment]
@ice-rahul I just merged `UpcomingRoomsCard`

**Comment by benawad** [comment]
@yvesnrb go for it

**Comment by yvesnrb** [comment]
> @yvesnrb go for it

On it

**Comment by ice-rahul** [comment]
@benawad starting `ProfileBlock`, thanks

**Comment by juliankrieger** [comment]
@benawad Are there any more planned?

**Comment by juliankrieger** [comment]
A couple of cards are missing (optional?) close buttons, for example the UserSummaryCard

**Comment by pr1m0x** [comment]
Are some components missing?
